Libraries/xcsdk/Tools/xcrun.cpp

```diff
@@ -295,94 +295,118 @@ main(int argc, char **argv)
         fprintf(stderr, "verbose: using developer root '%s'\n", manager->path().c_str());
     }
 
+    bool showSDKValue = options.showSDKPath() ||
+        options.showSDKVersion() ||
+        options.showSDKBuildVersion() ||
+        options.showSDKPlatformPath() ||
+        options.showSDKPlatformVersion();
+
     /*
      * Determine the SDK to use.
      */
-    xcsdk::SDK::Target::shared_ptr target = manager->findTarget(SDK.value_or("macosx"));
-    if (target == nullptr) {
-        if (SDK) {
-            fprintf(stderr, "error: unable to find sdk '%s'\n", SDK->c_str());
-        } else {
-            fprintf(stderr, "error: unable to find default sdk\n");
+    const std::string defaultSDK = "macosx";
+    xcsdk::SDK::Target::shared_ptr target = nullptr;
+    if (SDK) {
+        target = manager->findTarget(*SDK);
+        if (target == nullptr) {
+             printf("error: unable to find sdk: '%s'\n", SDK->c_str());
+             return -1;
+        }
+    } else {
+        target = target = manager->findTarget(defaultSDK);
+        /* nullptr target is not an error (except later on if SDK information is requested) */
+        if (showSDKValue && target == nullptr) {
```

Comment:
Hmm maybe move this down so the target check is closer to where it matters that target is not `nullptr`?